code/controllers/subsystem/blackmarket.dm

```diff
@@ -0,0 +1,166 @@
+
+				queued_purchases -= purchase
+				pad.add_to_queue(purchase)
+			// Get random area, throw it somewhere there.
+			if(SHIPPING_METHOD_TELEPORT)
+				var/turf/targetturf = get_safe_random_station_turf()
+				// This shouldn't happen.
+				if (!targetturf)
+					continue
+
+				// See comment for the same snipet above
+				if(ishuman(purchase.uplink.loc))
+					to_chat(purchase.uplink.loc, "<span class='notice'>[purchase.uplink] flashes a message noting that the order is being teleported to [get_area(targetturf)] in 60 seconds.</span>")
+				else
+					purchase.uplink.visible_message("<span class='notice'>[purchase.uplink] flashes a message noting that the order is being teleported to [get_area(targetturf)] in 60 seconds.</span>")
+
+				// do_teleport does not want to teleport items from nullspace, so it just forceMoves and does sparks.
+				addtimer(CALLBACK(src, /datum/controller/subsystem/blackmarket/proc/fake_teleport, purchase.entry.spawn_item(), targetturf), 60 SECONDS)
+				queued_purchases -= purchase
+				qdel(purchase)
+			// Get the current area of the uplink and drop the item there. Not in use currently (3/1/2020)
+			if(SHIPPING_METHOD_DROPPOD)
+				// This is mostly just copied from express cargo console but that should be fine.
+				var/area/A = get_area(purchase.uplink)
+				var/LZ
+				if(A.valid_territory)
+					var/list/empty_turfs
+					for(var/turf/open/floor/T in A.contents)
+						if(is_blocked_turf(T))
+							continue
+						LAZYADD(empty_turfs, T)
+						if(MC_TICK_CHECK)
+							break
+					if(empty_turfs && empty_turfs.len)
+						LZ = pick(empty_turfs)
+				// Bad area most likely, just drop it on the uplink. Sure this makes it just go to space and buy but w/e.
+				else
+					LZ = get_turf(purchase.uplink)
+
+				if(LZ)
+					new /obj/effect/DPtarget(LZ, supplypod_type, purchase.entry.spawn_item())
+
+
+					// See comment for the same snipet above
+					if(ishuman(purchase.uplink.loc))
+						to_chat(purchase.uplink.loc, "<span class='notice'>[purchase.uplink] flashes a message noting that the order is being dropped to [get_area(LZ)].</span>")
+					else
+						purchase.uplink.visible_message("<span class='notice'>[purchase.uplink] flashes a message noting that the order is being dropped to [get_area(LZ)].</span>")
+
+					queued_purchases -= purchase
+					qdel(purchase)
+
+					continue
+
+				// Guessing the uplink got sent to the void realm.
+				queued_purchases -= purchase
+				qdel(purchase)
+			// Get the current location of the uplink if it exists, then throws the item from space at the station from a random direction.
+			if(SHIPPING_METHOD_LAUNCH)
+				var/startSide = pick(GLOB.cardinals)
+				var/pickedloc = spaceDebrisStartLoc(startSide, purchase.uplink.z)
+
+				var/atom/movable/item = purchase.entry.spawn_item(pickedloc)
+				item.throw_at(purchase.uplink, 3, 3, spin = FALSE)
```

Comment:
This isnt working. The things end up nullspaced, you need to find out what pickedloc ends up being. 8/8 failed.

Reply:
Yeah, this was stupid and now it is fixed.